api/app/src/external/commonwell/document/shared.ts

```diff
@@ -1,39 +1,86 @@
-import { Document, DocumentContent } from "@metriport/commonwell-sdk";
+import { Document } from "@metriport/commonwell-sdk";
 import mime from "mime-types";
-import { MedicalDataSource } from "../..";
-import { DocumentReferenceCreate } from "../../../domain/medical/document-reference";
+import { DocumentReference } from "@medplum/fhirtypes";
 import { Patient } from "../../../models/medical/patient";
+import { Organization } from "../../../models/medical/organization";
 import { makePatientOID } from "../../../shared/oid";
 
 // TODO #340 When we fix tsconfig on CW SDK we can remove the `Required` for `id`
-export type DocumentWithLocation = Required<Pick<Document, "id">> &
-  Omit<DocumentContent, "location"> &
-  Required<Pick<DocumentContent, "location">> & {
-    fileName: string;
-    raw?: unknown;
-  };
+export type DocumentWithFilename = Document & {
+  fileName: string;
+};
+
+export const toFHIR = (
+  doc: DocumentWithFilename,
+  organization: Organization,
+  patient: Patient
+): DocumentReference => {
+  const id = doc.id?.replace("urn:uuid:", "");
 
-export function toDomain(patient: Patient) {
-  return (doc: DocumentWithLocation): DocumentReferenceCreate => {
-    return {
-      cxId: patient.cxId,
-      patientId: patient.id,
-      source: MedicalDataSource.COMMONWELL,
-      externalId: doc.id,
-      data: {
-        fileName: doc.fileName,
-        location: doc.location,
-        description: doc.description,
-        status: doc.status,
-        indexed: doc.indexed,
-        mimeType: doc.mimeType,
-        size: doc.size,
-        type: doc.type,
+  return {
+    id: id,
+    resourceType: "DocumentReference",
+    contained: [
+      {
+        resourceType: "Organization",
+        id: organization.id,
+        name: organization.data.name,
+      },
+      {
+        resourceType: "Patient",
+        id: patient.id,
+      },
+    ],
+    masterIdentifier: {
+      system: doc.content?.masterIdentifier?.system,
+      value: doc.content?.masterIdentifier?.value,
+    },
+    identifier: doc.content?.identifier?.map(id => {
+      return {
+        system: id.system,
+        value: id.value,
+        // question
+        // use: id.use,
+      };
+    }),
+    date: doc.content?.indexed,
+    status: "current",
+    type: doc.content?.type,
+    subject: {
+      reference: `Patient/${patient.id}`,
+      type: "Patient",
+    },
+    author: [
+      {
+        reference: `#${organization.id}`,
+        type: "Organization",
       },
-      raw: doc.raw,
-    };
+    ],
+    description: doc.content?.description,
+    content: [
+      {
+        attachment: {
+          title: doc.fileName,
+          contentType: "application/xml",
+          url: doc.content?.location,
+          extension: [
+            {
+              valueString: "S3",
+            },
+          ],
+        },
+      },
+    ],
+    extension: [
+      {
+        valueReference: {
+          reference: "Commonwell",
```

Comment:
nit: CommonWell -> we should probably define this string in a const somewhere, so we can use it throughout the application